testdata/project-v4/api/v1/captain_types.go

```diff
@@ -33,6 +33,11 @@ type CaptainSpec struct {
 	// foo is an example field of Captain. Edit captain_types.go to remove/update
 	// +optional
 	Foo *string `json:"foo,omitempty"`
+
+	// Bar is an experimental field that requires the "experimental-bar" feature gate to be enabled
+	// +feature-gate experimental-bar
+	// +optional
+	Bar *string `json:"bar,omitempty"`
```

Comment:
All samples are auto-generated.
Whenever we make changes, we run make install and make generate, which calls this script to regenerate the samples:
https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/generate.sh

Because of this, we can’t edit the samples directly. If the samples don’t match, the testdata CI check will fail.

For e2e tests, we create mock scenarios within the tests themselves. For example:
https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/alphagenerate/generate_test.go#L36-L62